src/librustdoc/clean/types.rs

```diff
@@ -1083,7 +1080,11 @@ impl Attributes {
     /// Finds all `doc` attributes as NameValues and returns their corresponding values, joined
     /// with newlines.
     crate fn doc_value(&self) -> Option<String> {
```

Comment:
not necessarily in this pr, but it would be nice to remove the `Option` here

Reply:
I think we check the `is_some` to generate differently the doc block. From my point of view, this forces us to keep this check so might be better to keep it.

Reply:
I don't really understand why `None` is treated differently from `Some("")` though. What if someone writes `#[doc = ""]`? Won't that return `Some("")` instead of `None`? And why do we need to special-case the absence of doc attrs so much?

Reply:
Another difference between `doc_value` and `collapsed_doc_value` - `doc_value` returns None if the *combined* string is empty, `collapsed_doc_value` returns None if there are no *lines*. The difference matters if there's empty lines, like below:
```
///
///
///
pub fn foo() {}
```

Reply:
@CraftSpider I see the following comment in `rustdoc-json-types`:
```
    /// The full markdown docstring of this item. Absent if there is no documentation at all,
    /// Some("") if there is some documentation but it is empty (EG `#[doc = ""]`).
    pub docs: Option<String>,
```
Is that an important distinction? Or can I just make that field a simple `String`?